decidim-core/lib/decidim/core/test/factories.rb

```diff
@@ -162,6 +166,35 @@
                role: :admin
       end
     end
+
+    trait :process_collaborator do
+      transient { participatory_process nil }
+
+      after(:create) do |user, evaluator|
+        create :participatory_process_user_role,
+               user: user,
+               participatory_process: evaluator.participatory_process,
+               role: :collaborator
+      end
+    end
+
+    trait :process_moderator do
+      transient { participatory_process nil }
+
+      after(:create) do |user, evaluator|
+        create :participatory_process_user_role,
+               user: user,
+               participatory_process: evaluator.participatory_process,
+               role: :moderator
+      end
+    end
+
```

Comment:
The moderator one will be easy to add. We should just create shared examples for the moderation and add tests for a participatory process moderator 😄 .